scripts/zones/Aht_Urhgan_Whitegate/npcs/Yahsra.lua

```diff
@@ -91,10 +91,13 @@ function onEventFinish(player,csid,option)
             else
                 return;
             end
-
-            player:addItem(itemID);
-            player:messageSpecial(ITEM_OBTAINED,itemID);
-            player:delCurrency("LEUJAOAM_ASSAULT_POINT",price);
+            if player:getFreeSlotsCount() > 0 then
```

Comment:
if you require scripts/globals/npc_util, lines 94-100 can be replaced with 
````
if npcUtil.giveItem(player, itemID) then
    player:delCurrency("LEUJAOAM_ASSAULT_POINT", price);
end
````

Reply:
Wiggo is right.  npcUtil replaces the code as written, but after checking wiki for what this NPC does, I agree that neither solution does what we want.

1. what is the correct message?

2. I don't see an easy way in LUA to tell whether a given itemId is rare or not.  This is important for cases like this, where some of the rewards are rare, and others not.

Reply:
@wrenffxi  will npcUtilis work for this now? i saw you updated it. does it require any other lines?

Reply:
Yes.  If npcUtil.giveItem(player, itemId) where itemId is a rare item the player already possesses, it will return false, which will prevent the deletion of currency.  The code I quoted in comment above should work now.